### ISSUE TITLE
Add custom title text for the iframe

### ISSUE THREAD

**Issue by colinaut** [gfscott/eleventy-plugin-embed-everything#205]
This is an enhancement request to add the ability to have custom title text on the iframe. Accessibly requirements at my institution encourage adding a title attribute on any iframe for screen readers (see [SiteImprove's Guidelines](https://help.siteimprove.com/support/solutions/articles/80001051799-accessibility-rule-inline-frame-without-a-text-alternative-explained)). Not sure how you'd implement it with this plugin but it would be great if possible.

As I assume you know, the normal markdown for an image allows titles for images as such:

```
![alt text](image url "image Title")
```

So maybe it could detect a single line with the title quoted after?

```
https://www.youtube.com/watch?v=TXh1zfd21T8 "iframe title"
```

Anyways, thanks for this plugin as it's been a great help! If you had the ability to add this it would be really useful.

**Comment by gfscott** [comment]
Cool! Yeah, I think this is doable and the accessibility improvement is definitely worthwhile.

I'll look into it a little more but I think my preferred way to handle this is to add a configurable option to download the video title via oEmbed ([example data](https://www.youtube.com/oembed?url=https://www.youtube.com/watch?v=TXh1zfd21T8)).

Just thinking out loud here:

**Pros:**
- Doesn't break the existing public API (currently any paragraphs with trailing characters after the YouTube URL [aren't processed](https://github.com/gfscott/eleventy-plugin-embed-everything/blob/main/packages/youtube/test/spotPattern.test.js#L66-L96) — but thinking about providing inline config options in 2.x someday...)
- Doesn't require the user to remember any syntax or type anything extra
- Configure once and it works indefinitely

**Cons:**
- Introduces a network call for every embed processed (if you turn it on)
- Opt-in requires you to configure it in the first place

On balance, I think making it opt-in globally is the right tradeoff. Interested if anyone has other perspectives to add!

**Comment by gfscott** [comment]
Re-opening this issue because adding this feature with v1.18.0 also introduced a bug that caused duplicate embeds. I "fixed" this with 1.18.1 by simply removing the feature. Hoping to add it back in soon.
